realm/kotlin-extensions/src/androidTest/kotlin/io/realm/KotlinRealmModelTests.kt

```diff
@@ -0,0 +1,233 @@
+package io.realm
+
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import io.realm.entities.PrimaryKeyClass
+import io.realm.entities.SimpleClass
+import io.realm.kotlin.*
+import io.realm.rule.RunInLooperThread
+import io.realm.rule.RunTestInLooperThread
+import io.realm.rule.TestRealmConfigurationFactory
+import org.junit.*
+import org.junit.Assert.*
+import org.junit.runner.RunWith
+
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+class KotlinRealmModelTests {
+
+    @Suppress("MemberVisibilityCanPrivate")
+    @get:Rule
+    val configFactory = TestRealmConfigurationFactory()
+
+    @get:Rule
+    val looperThread = RunInLooperThread()
+
+
+    private lateinit var realm: Realm
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getTargetContext())
+        realm = Realm.getInstance(configFactory.createConfiguration())
+    }
+
+    @After
+    fun tearDown() {
+        realm.close()
+    }
+
+    @Test
+    fun deleteFromRealm() {
+
+        // Make sure starting with 0
+        Assert.assertEquals(0, realm.where<SimpleClass>().count())
+
+        // Add 1, check count
+        realm.executeTransaction { it.createObject<SimpleClass>() }
+        Assert.assertEquals(1, realm.where<SimpleClass>().count())
+
+        // Delete the first, check count again.  !! is intentional to make
+        // sure we are sure calling deleteFromRealm
+        realm.executeTransaction { realm.where<SimpleClass>().findFirst()!!.deleteFromRealm() }
+        Assert.assertEquals(0, realm.where<SimpleClass>().count())
+
+    }
+
+    @Test
+    fun isValid() {
+        realm.executeTransaction {
+            val simple = it.createObject<SimpleClass>()
+            assertTrue("Expected valid after insert", simple.isValid())
+
+            simple.deleteFromRealm()
+            assertFalse("Expected invalid after delete", simple.isValid())
+        }
+    }
+
+   @Test
+    fun isManaged() {
+        realm.executeTransaction {
+            var simple = SimpleClass()
+            assertFalse("Expected not managed until attached", simple.isManaged())
+
+            simple = it.copyToRealm(simple)
+            assertTrue("Expected managed after attaching", simple.isManaged())
+        }
+    }
+
+    @Test
+    @RunTestInLooperThread
+    fun add_realmObjectChangeListener() {
```

Comment:
Generally, we name tests after the following pattern: `<methodUnderTest>_<conditions>`, so this would be `addObjectChangeListener_addObject` or something like that.